Feature/multi call roles

azizazlan opened the pull request:
Now with multicall to grant and revoke roles.

azizazlan committed:
only task castvote

azizazlan committed:
Updated gitignore for typescript

azizazlan committed:
Now understand a bit on quorom and delegate

azizazlan committed:
Fix grant roles and added queue and execute

azizazlan committed:
Changed total supply of token from 1000000 to 100 just to check the impact of quorum

azizazlan committed:
Rename and simplify helper functions

azizazlan committed:
Updated gitignore to ignore vim swap temp files

azizazlan committed:
Error with multicall

azizazlan committed:
Multicall now works

azizazlan committed:
Change encode functions name

PatrickAlphaC commented:
Love the suggestion here! Thank you!

But could we fix the PR so it's easier to review? Most of the changes look like they could be in a `.gitignore`

azizazlan commented:
OK I have fixed the formatting -now with 4 spaces tab width. (I was using 2 spaces tab width).

Sorry I am trying to understand your comment:
_Most of the changes look like they could be in a `.gitignore`_

Anyway I added `artifacts` and `[._]*.s[a-v][a-z]` in the file because I am using vim as the editor.

Let me know.

azizazlan committed:
Reformatted from 2 spaces to 4 spaces tab width

PatrickAlphaC commented:
Right. Sorry it's been hard to see what your changes are doing.

I think this one too is just looking to implement this with hardhat? Could you check this repo and see if it has everything you need? 

https://github.com/PatrickAlphaC/dao-template

azizazlan commented:
Thank you for replying my request. I will check it out and let you know.

On Mon, 21 Mar 2022 at 7:52 PM, Patrick Collins ***@***.***>
wrote:

> Right. Sorry it's been hard to see what your changes are doing.
>
> I think this one too is just looking to implement this with hardhat? Could
> you check this repo and see if it has everything you need?
>
> https://github.com/PatrickAlphaC/dao-template
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/brownie-mix/dao-mix/pull/4#issuecomment-1073802339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFSXJ6EKEGAMDWAREUV4QLVBBPIBANCNFSM5RBR2YWQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

azizazlan commented:
Below are your comments in the script file `deploy_and_run.py` 

```
    # Now, we set the roles...
    # Multicall would be great here ;)
```
So, I was trying to do multicall to set the roles and at the same time do it hardhat task.

What I should have done is implement it in brownie/python script - this would be a clearer "pull request".

I learnt my mistake and sorry I make you confuse.

PatrickAlphaC commented:
Ahhhhh, yes exactly!
